scripts/jenkins/ardana/gerrit/build_test_package.py

```diff
@@ -158,6 +169,50 @@ def _create_test_project(self):
                    self.obs_test_project_name, '--accept-in-hours', 720,
                    '-m', 'Auto delete after 30 days.')
 
+    def get_obsinfo_filename(self, package, service_def=None):
+        if not service_def:
+            service_def_cmd = sh.osc(
+                '-A', 'https://api.suse.de', 'cat',
+                self.obs_linked_project,
+                package.name,
+                '_service')
+            service_def = str(service_def_cmd)
+        root = ET.fromstring(service_def)
+        nodes = root.findall(
+            './service[@name="obs_scm"]/param[@name="filename"]')
+        if len(nodes) != 1 or not nodes[0].text:
+            raise ValueError(
+                "There needs to be exactly one obs_scm service filename"
+                " in https://build.suse.de/package/view_file/%s/%s/_service"
+                % (self.obs_linked_project, package.name))
+        return nodes[0].text
+
+    def is_current(self, package):
+        if package.has_applied_changes():
+            return False
+
+        obsinfo_filename = self.get_obsinfo_filename(package)
+        obsinfo = sh.osc(
+            '-A', 'https://api.suse.de', 'cat',
+            self.obs_linked_project,
+            package.name,
+            '%s.obsinfo' % obsinfo_filename)
+
+        matches = re.findall('^commit: (\S+)$', str(obsinfo), re.MULTILINE)
+        if len(matches) != 1:
+            raise ValueError(
+                "There needs to be exactly one commit value"
+                " in https://build.suse.de/package/view_file/%s/%s/%s.obsinfo"
+                % (
+                self.obs_linked_project, package.name, obsinfo_filename))
+        current_commit = matches[0]
+
+        gerrit_query = "/projects/ardana%2F{}/branches/{}".format(
```

Comment:
Would it not be more readable/maintainable/re-usable to move this to a separate (helper) method? E.g. target_branch_head() taking package as an argument...